packages/fxa-content-server/app/scripts/views/pair/index.js

```diff
@@ -26,17 +28,23 @@ class PairIndexView extends FormView {
       return this.replaceCurrentPage('pair/unsupported');
     }
 
+    const account = this.broker.get('browserSignedInAccount');
     // If we reach this point that means we are in Firefox Desktop
-    if (! this.broker.get('browserSignedInAccount')) {
-      // if we are not logged into Sync then we offer to sign in
+    if (! account) {
+      // if we are not logged into Sync then we offer to connect
       return this.replaceCurrentPage('connect_another_device');
     }
 
+    if (! account.verified || ! account.sessionToken) {
+      // if account is not verified or missing sessionToken then offer to sign in or confirm
+      return window.location.href = this.getEscapedSyncUrl('signin', 'fxa:pair');
```

Comment:
Can you do a `this.navigateAway` with the URL here so that metrics are flushed before redirecting?